make crashes caused by missing /etc/avahi/aliases

I tried to install _avahi-aliases_ but `make install` crashed  in line 15, because `/etc/avahi/aliases` has not been created in the installation process (in the [original version](https://github.com/airtonix/avahi-aliases) and the other pip flavored solutions the config files are coppied from the package to the file system).

```
…
>>> Installing systemd unit
sudo cp etc/avahi-alias.service /etc/systemd/system/
sudo systemctl daemon-reload
>>> Enabling systemd unit
sudo systemctl enable avahi-alias.service
Created symlink /etc/systemd/system/multi-user.target.wants/avahi-alias.service → /etc/systemd/system/avahi-alias.service.
sudo systemctl restart avahi-alias.service
Job for avahi-alias.service failed because the control process exited with error code.
See "systemctl status avahi-alias.service" and "journalctl -xe" for details.
make: *** [Makefile:15: install-unit] Fehler 1
```

O.k., let's look what happened:

```
thosch@testbed:~/avahi-aliases-master $ systemctl status avahi-alias.service
● avahi-alias.service - Avahi Aliases
   Loaded: loaded (/etc/systemd/system/avahi-alias.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Mon 2020-05-25 12:18:09 BST; 20s ago
  Process: 2868 ExecStart=/usr/local/bin/avahi-publish-aliases (code=exited, status=1/FAILURE)

Mai 25 12:18:08 testbed systemd[1]: Starting Avahi Aliases...
Mai 25 12:18:09 testbed avahi-publish-aliases[2868]: Traceback (most recent call last):
Mai 25 12:18:09 testbed avahi-publish-aliases[2868]:   File "/usr/local/bin/avahi-publish-aliases", line 19, in <module>
Mai 25 12:18:09 testbed avahi-publish-aliases[2868]:     alias_file = open(alias_file_path)
Mai 25 12:18:09 testbed avahi-publish-aliases[2868]: FileNotFoundError: [Errno 2] No such file or directory: '/etc/avahi/aliases'
Mai 25 12:18:09 testbed systemd[1]: avahi-alias.service: Control process exited, code=exited, status=1/FAILURE
Mai 25 12:18:09 testbed systemd[1]: avahi-alias.service: Failed with result 'exit-code'.
Mai 25 12:18:09 testbed systemd[1]: Failed to start Avahi Aliases.
```

Let's try to add an alias, the file should be generated…

```
thosch@testbed:~/avahi-aliases-master $ avahi-add-alias testhost.local
Traceback (most recent call last):
  File "/usr/local/bin/avahi-add-alias", line 8, in <module>
    alias_file = open("/etc/avahi/aliases")
IOError: [Errno 2] No such file or directory: '/etc/avahi/aliases'
```

Ooops! 😮 

After _manually_ adding `/etc/avahi/aliases` the second run of `make install` worked fine.

--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@ deps:
 install: deps install-binaries install-unit
 
 install-binaries:
-	touch /etc/avahi/aliases
+	sudo touch /etc/avahi/aliases
 	sudo cp bin/avahi* /usr/local/bin
 
 install-unit:

--- a/Makefile
+++ b/Makefile
@@ -4,6 +4,7 @@ deps:
 install: deps install-binaries install-unit
 
 install-binaries:
+	touch /etc/avahi/aliases
 	sudo cp bin/avahi* /usr/local/bin
 
 install-unit:

--- a/Makefile
+++ b/Makefile
@@ -8,14 +8,14 @@ install-binaries:
 	sudo cp bin/avahi* /usr/local/bin
 
 install-unit:
-	@echo ">>> Installing systemd unit"
+	@echo "\n>>> Installing systemd unit"
 	sudo cp etc/avahi-alias.service /etc/systemd/system/
 	sudo systemctl daemon-reload
-	@echo ">>> Enabling systemd unit"
+	@echo "\n>>> Enabling systemd unit"
 	sudo systemctl enable avahi-alias.service
 	sudo systemctl restart avahi-alias.service
 	sudo systemctl status avahi-alias.service
-	@echo ">>> Install done.\n Add aliases to /etc/avahi/aliases or /etc/avahi/aliases.d/* and use\n    sudo systemctl <start|stop|status> avahi-alias.service\n"
+	@echo "\n>>> Install done.\n Add aliases to /etc/avahi/aliases or /etc/avahi/aliases.d/* and use\n    sudo systemctl <start|stop|status> avahi-alias.service\n"
 
 uninstall-unit:
 	@echo ">>> Uninstallling systemd unit"

--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 deps:
-	sudo apt install python-avahi python-dbus
+	sudo apt install -y python-avahi python-dbus
 
 install: deps install-binaries install-unit
 

--- a/bin/avahi-add-alias
+++ b/bin/avahi-add-alias
@@ -1,4 +1,5 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
+
 import os, sys
 from subprocess import Popen, PIPE
 

--- a/bin/avahi-list-aliases
+++ b/bin/avahi-list-aliases
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import os, sys
 from subprocess import Popen

--- a/bin/avahi-list-aliases
+++ b/bin/avahi-list-aliases
@@ -23,4 +23,4 @@ for line in alias_file:
         aliases.add(entry)
 
 for alias in aliases:
-    print alias 
+    print(alias)

--- a/bin/avahi-publish-aliases
+++ b/bin/avahi-publish-aliases
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env python2
 
 import os, sys
 from subprocess import Popen

--- a/bin/avahi-publish-aliases
+++ b/bin/avahi-publish-aliases
@@ -10,20 +10,18 @@ def ensure_file(path):
         rfile = open(path,"w+");
     return rfile
 
-
 command = '/usr/local/bin/avahi-publish-domain-alias'
 alias_pid_path = "/tmp/avahi-publish-alias.pid"
 alias_file_path = "/etc/avahi/aliases"
 alias_dir_glob = "/etc/avahi/aliases.d/*"
 
 alias_file = open(alias_file_path)
+
 if not os.path.exists(alias_pid_path):
     open(alias_pid_path,"w").close()
 
 alias_dir = glob.glob(alias_dir_glob)
-
 alias_pid = open(alias_pid_path,"r")
-
 aliases = set()
 
 for line in alias_pid:

--- a/bin/avahi-publish-domain-alias
+++ b/bin/avahi-publish-domain-alias
@@ -1,5 +1,4 @@
-#! /usr/bin/env python
-# avahi-alias.py
+#!/usr/bin/env python2
 
 import avahi, dbus
 from encodings.idna import ToASCII

--- a/bin/avahi-remove-alias
+++ b/bin/avahi-remove-alias
@@ -1,4 +1,5 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
+
 import os,sys
 from subprocess import Popen, PIPE
 import glob

--- a/bin/avahi-remove-alias
+++ b/bin/avahi-remove-alias
@@ -12,6 +12,7 @@ alias_dir = glob.glob(alias_dir_glob)
 for alias in args:
     alias_file = open("/etc/avahi/aliases")
     alias_output = []
+
     for line in alias_file:
         line = line.rstrip("\n")
         if line != alias:

--- a/bin/avahi-remove-alias
+++ b/bin/avahi-remove-alias
@@ -13,6 +13,7 @@ for alias in args:
     alias_file = open("/etc/avahi/aliases")
     alias_output = []
     for line in alias_file:
+        line = line.rstrip("\n")
         if line != alias:
             alias_output.append(line)
 

